### PR TITLE
Moving old xontribs to repositories

### LINKED ISSUE #5051: Separating xonsh/xontrib to repositories
We have 11 xontribs in [xonsh/xontrib](https://github.com/xonsh/xonsh/tree/main/xontrib). 

If we separate the xontribs to repositories:
* We will have [almost 100 xontribs on Github](https://github.com/topics/xontrib)
* We will have separate rating, description and issues like in [mpl, jupyter, jedi, etc](https://github.com/xonsh?q=xontrib-&type=all&language=&sort=)

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

### PR DESCRIPTION
**Pull request by anki-code**
Closes #5051

TODO:

- [x] Create xonsh/xontrib-* repositories
- [x] Transfer xonitrb issues from xonsh/issues to xonsh/xonitrb-*/issues
- [x] Transfer abbrevs to xonsh/xontrib-abbrevs
- [x] Transfer bashisms to xonsh/xontrib-bashisms
- [x] Transfer free_cwd to xonsh/xontrib-free_cwd
- [x] Transfer whole_word_jumping to xonsh/xontrib-whole-word-jumping
- [x] Transfer fish_completer to xonsh/xontrib-fish-completer
- [x] Transfer vox, autovox, voxapi to xonsh/xontrib-vox
- [x] Transfer pdb, xog to xonsh/xontrib-debug-tools
- [x] Keep coreutils xontrib in xonsh/xonsh
- [x] Add xontrib installation instructions to docs
- [x] Recheck config for tests - https://github.com/xonsh/xontrib-template/issues/24
- [x] Upload packages to PyPi

Next steps:

* ~[Update awesome-xontribs](https://github.com/xonsh/awesome-xontribs/pull/13)~ (done)
* [Implement adding the abbrevs section into `xonfig web`](https://github.com/xonsh/xontrib-abbrevs/issues/2).

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

### CODE REVIEW

**Comment by jnoortheen** [comment]
@anki-code I wanted to move xontribs which depended on other 3rd party libraries. Only ptk-shelll and abbrevs remain under xontribs of that sort; others were moved out.
One suggestion though, Instead of hello world, we can leave some plugins that is minimal and robust (doesn't need much changes). Also I think the code from `xontrib-coreutils` is used inside `xonsh` .

**Comment by jnoortheen** [comment]
> bashisms transfered to https://github.com/xonsh/xontrib-bashisms

Could you consider removing `requirements.txt/setup.py` while using [PEP-621](https://github.com/xonsh/xontrib-template#advent-of-pep-621) ?

**Comment by jnoortheen** [comment]
> Just note: while I worked with abbrevs I found the Routes code for `xonfig web`. It's very cool to have an ability to enrich web config! I think the same should be available for `xonfig` output as well.

You can check the xonfig module code. It already has a [plugin sort of interface](https://github.com/xonsh/xonsh/blob/main/xonsh/xonfig.py#L711) to load additional shells and list them.

**Comment by gforsyth** [comment]
> Could you please add `PYPI_API_TOKEN` to the new repositories:

An API token for what?  These packages don't exist yet and I can't scope an API token for my entire PyPI account and stick it in a github secret.

**Comment by jnoortheen** [comment]
@anki-code I have checked the new repos.  there is still `setup.py` in those, it is not needed .

File: docs/python_virtual_environments.rst [hunk 1]
```diff
@@ -8,7 +8,9 @@ Python Virtual Environments
 
 The usual tools for creating Python virtual environments—``venv``, ``virtualenv``, ``pew``—don't play well with xonsh. We won't dig deeper into why it is so, but the general gist is that these tools are hacky and hard-coded for bash, zsh, and other mainstream shells.
 
-Luckily, xonsh ships with its own virtual environments manager called **Vox**.
+Luckily, xonsh ships with its own virtual environments manager called **Vox**. Run to install Vox::
```
Comment (jnoortheen) [inline docs/python_virtual_environments.rst:11]: instead of ships, it can be changed to `has`

File: README.rst [hunk 2]
```diff
@@ -55,9 +55,8 @@ And visit https://xon.sh for more information:
 
 Extensions
 **********
-- `Core extensions (xontribs) <https://xon.sh/api/_autosummary/xontribs/xontrib.html>`_
-- `External extensions on Github <https://github.com/topics/xontrib>`_
-- `List of awesome xontribs <https://github.com/xonsh/awesome-xontribs>`_
+- `Extensions on Github <https://github.com/topics/xontrib>`_
```
Comment (jnoortheen) [inline README.rst:58]: it still has some core xontribs.

**Comment by jnoortheen** [comment]
@anki-code stripping all the xontribs is fine. But does that mean, we no longer accept core xontribs and deprecate putting xontribs in the common folder ? I think , this needs some clarification and consensus among the community.

**Comment by gforsyth** [comment]
I'm a -0 on this.
The distinction to me was always that core xontribs were maintained by us.
This doesn't change that, since they're still in the org, just now we get a bunch more automated emails for pre-commit, etc.
But you've done the work, so I'm not going to block it -- we'll probably want to add some of the now unbundled xontribs to the conda-forge recipe so we don't break a bunch of people.

File: news/xontribs_transfer.rst [hunk 3]
```diff
@@ -0,0 +1,29 @@
+**Added:**
+
+* <news item>
+
+**Changed:**
+
+* ``abbrevs`` xontrib transferred to ``xonsh/xontrib-abbrevs``.
```
Comment (jnoortheen) [inline news/xontribs_transfer.rst:7]: please update the actual github repo links

**Review by jnoortheen — APPROVED** [review]

**Comment by bestlem** [comment]
> > to add some of the now unbundled xontribs to the conda-forge recipe so we don't break a bunch of people.
> 
> If the default package will have the same list of xontribs I don't like it. Personally I'm using only two ex-core xontribs and I don't want to install others. In the future I would like to create statistics by xontrib installations and the inclusion of the ex-core to default package boost the numbers syntatically.

This could be covered by having xonsh and xonsh[full] recipes as I think exist in some distributions and mentioned https://github.com/xonsh/xonsh/issues/5051#issuecomment-1429799195

**Comment by gforsyth** [comment]
Conda doesn't support extras in the same way as pip, I imagine we'll end up with multiple packages (which is fine). One with no xontribs except coreutils, and one with.. some number more of them. Also need to set up feedstocks for all of the unbundled xontribs if we want to do that (and maybe we don't)

@anki-code - thanks for all the work on this!
Your point about issue separation is a good one.

**Pull request merged**